include/zephyr/net/http/server.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_HTTP_SERVER_H_
+#define ZEPHYR_INCLUDE_NET_HTTP_SERVER_H_
+
+#if !defined(__ZEPHYR__) || defined(CONFIG_POSIX_API)
+
+#include <stdint.h>
+#include <poll.h>
+
+#else
+
+#include <zephyr/posix/poll.h>
+
+#endif
+
+#if !defined(__ZEPHYR__)
+
+#define CONFIG_NET_HTTP2_MAX_CLIENTS 10
+#define CONFIG_NET_HTTP2_MAX_STREAMS 5
+#define CONFIG_NET_HTTP2_CLIENT_BUFFER_SIZE 256
+
+#endif
+
+#define CLIENT_BUFFER_SIZE CONFIG_NET_HTTP2_CLIENT_BUFFER_SIZE
+#define MAX_CLIENTS CONFIG_NET_HTTP2_MAX_CLIENTS
+#define MAX_STREAMS CONFIG_NET_HTTP2_MAX_STREAMS
+
+struct http2_server_config {
+	int port;
+	int address_family;
+};
+
+enum http2_server_state {
+	INITIAL_STATE,
+	HTTP_PREFACE,
+	HTTP1_REQUEST,
+	HTTP2_FRAME_SETTINGS,
```

Comment:
And we should ensure there is no name collisions, I suggest to suffix all of the state enums with `_STATE`.